sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp

```diff
@@ -1221,7 +1221,7 @@ getRangeOfAllowedCopyEngines(const ur_device_handle_t &Device) {
   // immediate commandlists are being used. For standard commandlists all are
   // used.
   if (!EnvVar) {
-    if (Device->ImmCommandListUsed)
+    if (Device->useImmediateCommandLists())

```

Comment:
useImmediateCommandLists() will read again the env var. ImmCommandListUsed already has the value cached. Any reason to use useImmediateCommandLists()?

Reply:
Because the env var value is initialized using "static const int ImmediateCommandlistsSetting = ", the env var is read only once. The body of the lambda is executed only the first time.